ThirdpartyToolchain.cmake

```diff
@@ -125,8 +125,12 @@ macro(build_protobuf)
 
     check_cxx_compiler_flag("-Winvalid-noreturn"
                             COMPILER_HAS_W_INVALID_NORETURN)
+
     if(COMPILER_HAS_W_INVALID_NORETURN)
       string(APPEND CMAKE_CXX_FLAGS " -Wno-invalid-noreturn")
+    else()
```

Comment:
So are we saying in this case that the compiler fails, but doesn't give you flag to disable that specific check?

Reply:
Yes, the compiler raises a warning `'noreturn' function does return` but not the specific check and we error because of `-Werror`, see prior to the change:
```
[305/1452] Building CXX object _deps/protobuf-build/CMakeFiles/libprotobuf-lite.dir/src/google/protobuf/generated_message_tctable_lite.cc.o
FAILED: _deps/protobuf-build/CMakeFiles/libprotobuf-lite.dir/src/google/protobuf/generated_message_tctable_lite.cc.o 
ccache /usr/bin/c++ -DGOOGLE_PROTOBUF_CMAKE_BUILD -DHAVE_ZLIB -I/velox/_build/debug/_deps/protobuf-src/src -I/velox/_build/debug/_deps/protobuf-build -mavx2 -mfma -mavx -mf16c -mlzcnt -std=c++17 -mbmi2 -D USE_VELOX_COMMON_BASE -D HAS_UNCAUGHT_EXCEPTIONS -Werror -Wall -Wextra -Wno-unused        -Wno-unused-parameter        -Wno-sign-compare        -Wno-ignored-qualifiers        -Wno-implicit-fallthrough          -Wno-empty-body          -Wno-class-memaccess          -Wno-comment          -Wno-int-in-bool-context          -Wno-redundant-move          -Wno-array-bounds          -Wno-maybe-uninitialized          -Wno-unused-result          -Wno-format-overflow          -Wno-strict-aliasing          -Wno-type-limits -Wno-missing-field-initializers -Wno-stringop-overflow -g -fPIC -std=c++11 -MD -MT _deps/protobuf-build/CMakeFiles/libprotobuf-lite.dir/src/google/protobuf/generated_message_tctable_lite.cc.o -MF _deps/protobuf-build/CMakeFiles/libprotobuf-lite.dir/src/google/protobuf/generated_message_tctable_lite.cc.o.d -o _deps/protobuf-build/CMakeFiles/libprotobuf-lite.dir/src/google/protobuf/generated_message_tctable_lite.cc.o -c /velox/_build/debug/_deps/protobuf-src/src/google/protobuf/generated_message_tctable_lite.cc
In file included from /velox/_build/debug/_deps/protobuf-src/src/google/protobuf/generated_message_tctable_lite.cc:36:
/velox/_build/debug/_deps/protobuf-src/src/google/protobuf/generated_message_tctable_impl.h: In function 'void google::protobuf::internal::AlignFail(uintptr_t) [with long unsigned int align = 4]':
/velox/_build/debug/_deps/protobuf-src/src/google/protobuf/generated_message_tctable_impl.h:256:1: error: 'noreturn' function does return [-Werror]
  256 | }
      | ^
/velox/_build/debug/_deps/protobuf-src/src/google/protobuf/generated_message_tctable_impl.h: In function 'void google::protobuf::internal::AlignFail(uintptr_t) [with long unsigned int align = 8]':
/velox/_build/debug/_deps/protobuf-src/src/google/protobuf/generated_message_tctable_impl.h:256:1: error: 'noreturn' function does return [-Werror]
cc1plus: all warnings being treated as errors
```